Controllers/AbstractControllers/AbstractUserController.php

```diff
@@ -210,6 +247,579 @@ private function getPostInfo(): IDataManagement
         return $this->postInfo;
     }
 
+
+    public function profileAction(array $params = null): void
+    {
+        if (!$this->userModel->isSigned() || $this->validateRequester()) {
+            $this->redirectHome();
+
+            return;
+        }
+
+        try {
+            $this->getMsgModel();
+            if ($params !== null) {
+                throw new \Exception('Params have to be empty in profileAction!');
```

Comment:
So you throw the exception here in this function and catch it here as well? What is the purpose of this move?